pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,147 @@
+    sync(container_remote, container_repo)
 
-    @classmethod
-    def tearDownClass(cls):
-        """Delete things made in setUpClass. addCleanup feature does not work with setupClass."""
-        cls.repository_api.delete(cls.from_repo.pulp_href)
-        cls.remote_api.delete(cls.remote.pulp_href)
-        delete_orphans()
-
-    def test_sync_idempotency(self):
-        """Ensure that sync does not create orphan tags https://pulp.plan.io/issues/5252 ."""
-        sync_data = ContainerRepositorySyncURL(remote=self.remote.pulp_href)
-        sync_response = self.repository_api.sync(self.from_repo.pulp_href, sync_data)
-        monitor_task(sync_response.task)
-
-        tags_api = ContentTagsApi(self.client_api)
-        first_sync_tags_named_a = tags_api.list(name="manifest_a")
-
-        sync_response = self.repository_api.sync(self.from_repo.pulp_href, sync_data)
-        monitor_task(sync_response.task)
-
-        second_sync_tags_named_a = tags_api.list(name="manifest_a")
-
-        self.assertEqual(first_sync_tags_named_a.count, 1)
-        self.assertEqual(second_sync_tags_named_a.count, 1)
-
-
-class FilteredTagsSyncTestCase(PulpTestCase):
-    """A test case for syncing repositories with filtered tags."""
-
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        client_api = gen_container_client()
-        cls.repositories_api = RepositoriesContainerApi(client_api)
-        cls.remotes_api = RemotesContainerApi(client_api)
-        cls.tags_api = ContentTagsApi(client_api)
-        cls.versions_api = RepositoriesContainerVersionsApi(client_api)
+    second_sync_tags_named_a = container_tag_api.list(name="manifest_a")
 
-        cls.repository = None
+    assert first_sync_tags_named_a.count == 1
+    assert second_sync_tags_named_a.count == 1
 
-    def test_sync_with_non_existing_included_tag(self):
-        """Check whether the sync machinery ignores a non-existing tag."""
-        include_tags = ["manifest_a", "non_existing_manifest"]
-        self.sync_repository_with_filtered_tags(include_tags)
 
-        self.assert_synced_tags(["manifest_a"])
+@pytest.mark.parallel
+def test_sync_with_non_existing_included_tag(
+    synced_repository_with_filtered_tags,
+    container_repository_api,
+    container_tag_api,
+):
+    include_tags = ["manifest_a", "non_existing_manifest"]
+    synced_repo = synced_repository_with_filtered_tags(include_tags)
 
-    def test_sync_with_included_tags(self):
-        """Test whether the repository is synced only with included tags."""
-        include_tags = ["manifest_a", "manifest_b", "manifest_c"]
-        self.sync_repository_with_filtered_tags(include_tags)
+    latest_repo_version = container_repository_api.read(synced_repo.pulp_href).latest_version_href
+    tags = container_tag_api.list(repository_version=latest_repo_version).results
 
-        self.assert_synced_tags(include_tags)
+    assert sorted(["manifest_a"]) == sorted(tag.name for tag in tags)
```

Comment:
```suggestion
    assert ["manifest_a"] == list(tag.name for tag in tags)
```
Or maybe you can use `pytest.mark.parameterize` to collapse some these tests.

Reply:
I collapsed almost all similar tests, just left one which I think that makes sense to be there.